inlang/source-code/sdk/src/resolve-modules/cache.ts

```diff
@@ -0,0 +1,55 @@
+import type { NodeishFilesystemSubset } from "@inlang/plugin"
+import { type Result, tryCatch } from "@inlang/result"
+
+function escape(url: string) {
+	// collect the bytes of the UTF-8 representation & hex encode
+	const bytes = new TextEncoder().encode(url)
+	const escaped = [...bytes].map((byte) => byte.toString(16).padStart(2, "0")).join("")
+	return escaped
+}
+
+async function readModuleFromCache(
+	moduleURI: string,
+	projectPath: string,
+	readFile: NodeishFilesystemSubset["readFile"]
+): Promise<Result<string, Error>> {
+	const moduleHash = escape(moduleURI)
+	const filePath = projectPath + `/cache/${moduleHash}.js`
```

Comment:
We can't use the module ID in the file-name since we need to be able to know the file-name before loading the module. The module ID cannot always be determined from the URL.

Currently the "hash" is just a hashed version of the URL to make sure it's FS compatible. It doesn't have anything to do with the content.

Reply:
> We can't use the module ID in the file-name since we need to be able to know the file-name before loading the module. The module ID cannot always be determined from the URL.

makes sense.

Change the path from `/cache/*` to `/cache/modules/*` to enable `/cache/lint-reports/*` and other caches in the future?